### DISCUSSION TITLE
How to end http request with custom error codes like 400, 401 ?

### DISCUSSION THREAD

**Discussion by saikumarpb** [uNetworking/uWebSockets.js#824] [Q&A] [answered]
my current implementation , i am closing http as shown below

`res.writeStatus("401 Unauthorized").end(authFailureResponse())`

but this is being sent with http response status as  200

**Comment by e3dio** [comment, marked as answer]
You probably wrote a header before that so it auto writes status 200. Write status before other headers

**Reply by saikumarpb** [reply]
no , i dont have a header before that

```
app.post("/app/route", async (res, req) => {
    res.onAborted(() => {
      console.log(
        "Unhandled read error from Node.js, you need to handle this!"
      );
    });
    try {
      if( some conition ) 
      { 
           // some functionality
       } 
     else {
        console.log("Invalid Auth");
        res.cork(() => {
          res.writeStatus("401");
          res.end(authFailureResponse());
        });
      }
    } catch (err) {
      console.log(`Error on authenticating the private stream request: ${err}`);
    }
  });
```

**Reply by e3dio** [reply]
I ran exactly the code above and it works fine except your route string is bad needs to be `/app/route` not `app/route` so I'm guessing you did not try the code you just posted

![image](https://user-images.githubusercontent.com/85405955/199766005-b7ae104b-cc39-4edd-871d-101f4227087a.png)

**Reply by e3dio** [reply]
If you can't do basic troubleshooting and testing how can you expect other people to do it for you. You probably have your routes messed up or something, do some basic testing

**Reply by saikumarpb** [reply]
hey thanks for answering, the route in the above-posted code was a typo while posting it here. 

I have got the bug fixed . as you mentioned there was a function writing a header related to cors before writing the status. as this is done in the nested function I didn't catch it.

**Reply by uNetworkingAB** [reply]
I'm going to add things like these to the smoke test so we always test these common operations.
